### PR TITLE
Fix build for Apple Clang

### PR DESCRIPTION
**Pull request by jamierpond**
Had some type inference issues on Apple Clang, also cleaned up a few spots using `sprintf` as opposed to `snprintf`.

### CODE REVIEW

**Review by thecppzoo — APPROVED** [review]
Thanks!
This code is not completely operational, I assure you Scott and I have been working on the SWAR library to come back to finish this

**Comment by jamierpond** [comment]
Great! Pls lmk if there is anything else about this PR you'd like to see changed before you merge! Looks like I can't do it on my side :)

**Comment by thecppzoo** [comment]
> Great! Pls lmk if there is anything else about this PR you'd like to see changed before you merge! Looks like I can't do it on my side :)

Don't you have permission?
I went ahead and merged for you, please confirm whether you have merge permissions.

**Pull request merged**
